asv_bench/benchmarks/period.py

```diff
@@ -127,17 +54,78 @@ def time_start_time(self):
     def time_end_time(self):
         self.per.end_time
 
-    def time_to_timestamp():
+
+class PeriodUnaryMethods(object):
+    params = ['M', 'min']
+    param_names = ['freq']
+
+    def setup(self, freq):
+        self.per = Period('2012-06-01', freq=freq)
+
+    def time_to_timestamp(self):
         self.per.to_timestamp()
 
-    def time_now():
+    def time_now(self):
         self.per.now()
 
-    def time_asfreq():
+    def time_asfreq(self):
         self.per.asfreq('A')
 
 
-class period_standard_indexing(object):
+class PeriodIndexConstructor(object):
+    goal_time = 0.2
+
+    params = ['D']
+    param_names = ['freq']
+
+    def setup(self, freq):
+        self.freq = freq
+        self.rng = date_range('1985', periods=1000)
+        self.rng2 = date_range('1985', periods=1000).to_pydatetime()
+
+    def time_from_date_range(self):
+        PeriodIndex(self.rng, freq=self.freq)
+
+    def time_from_pydatetime(self):
+        PeriodIndex(self.rng2, freq=self.freq)
+
+
+class DataFramePeriodColumn(object):
+    goal_time = 0.2
+
+    def setup_cache(self):
+        rng = pd.period_range(start='1/1/1990', freq='S', periods=20000)
+        df = pd.DataFrame(index=range(len(rng)))
+        return rng, df
+
+    def time_setitem_period_column(self, tup):
+        rng, df = tup
+        df['col'] = rng
+
+
+class Algorithms(object):
+    goal_time = 0.2
+
+    params = [PeriodIndex, Series]
```

Comment:
what does this give in the asv run output as name for the benchmark? (just 'PeriodIndex' or the full path? in the last case I would just do `['index', 'series']`)

Reply:
> what does this give in the asv run output as name for the benchmark? 

I'm not sure.  Unfortunately I'm not aware of a quick-feedback way of running these.

Going to put a pin in this PR for a bit, much rather focus on offsets where current PRs need to get resolved before I can do actual bugfixes.

Reply:
Check 'asv dev', you can select only this file and specify a quick run

Reply:
Quickly checked this (with `asv dev -b period` and it used the full path, so please use `['index', 'series']` as the `params`